web/modules/custom/portland/modules/portland_location_picker/js/portland_location_picker.js

```diff
@@ -810,34 +810,65 @@
-
-          if (requireCityLimits && displayCityLimits) {
-            // check if click is within Portland city limits. if not, use js alert to show error message and return null.
-            var inLayer = leafletPip.pointInLayer(latlng, muniLayer, false);
-            if (inLayer.length > 0) {
-              var municipalityName = inLayer[0].feature.properties.CITYNAME;
-              if (municipalityName != "Portland") {
-                console.log("Clicked in " + municipalityName + ".");
-                showStatusModal(CITY_LIMITS_MESSAGE);
-                return false;
+          if (requireCityLimits || requireCityLimitsPlusParks) {
+            var success = false;
+            $.ajax({
+              async: false,
+              url: buildCheckCityLimitsFenceUrl(latlng),
+              success: function (cityBoundaryResponse) {
+                cityBoundaryResponse = JSON.parse(cityBoundaryResponse);
+                if (!cityBoundaryResponse.features || cityBoundaryResponse.features.length < 1 || cityBoundaryResponse.features[0].attributes.CITYNAME.toLowerCase() != "portland") {
+                  // not in city boundary. if configured, try parks boundaries.
+                  if (requireCityLimitsPlusParks) {
+                    $.ajax({
+                      async: false,
+                      url: buildCheckCityLimitsPlusParksFenceUrl(latlng),
+                      success: function (parksBoundaryResponse) {
+                        parksBoundaryResponse = JSON.parse(parksBoundaryResponse);
+                        if (!parksBoundaryResponse.features || parksBoundaryResponse.features.length < 1) {
+                          // not in parks boundary either, display error message
+                          showStatusModal(CITY_LIMITS_MESSAGE);
+                          success = false;
+                        } else {
+                          success = true;
+                        }
+                      },
+                      error: function (e) {
+                        // Handle any error cases
+                        showErrorModal(response.error.message);
+                      }
+                    });
+                  }
+                } else {
+                  // is within city boundary
+                  success = true;
+                }
+              },
+              error: function (e) {
+                // Handle any error cases
+                showErrorModal(response.error.message);
               }
-            } else {
-              console.log("Clicked other area.");
-              console.log('Coordinates:' + latlng[0] + ', ' + latlng[1])
-              showStatusModal(CITY_LIMITS_MESSAGE);
-              return false;
-            }
-
-            $('#location_is_portland').val("Yes");
-            console.log("Is Portland: " + $('#location_is_portland').val());
-            $('#location_municipality_name').val(cityBoundaryLayer.municipality);
```

Comment:
It looks like this PR removed `location_is_portland` and `location_municipality_name`. If that was intentional, we should probably remove those from the composite element as well. It doesn't look like they're used in any form currently, FWIW.

Reply:
I've reinstated the fields. Might be useful in conditional logic in a future form.